packages/babel-parser/src/parser/expression.js

```diff
@@ -268,6 +270,12 @@ export default class ExpressionParser extends LValParser {
         }
         return left;
       }
+    } else if (
+      this.isContextual("match") &&
+      this.hasPlugin("patternMatching") &&
+      !this.hasFollowingLineBreak()
+    ) {
+      return this.parseMaybeMatchExpression();
```

Comment:
@ljharb Can I assume `match {}` is a primary expression?

If so we should handle it in `parseExprAtom`, like #13573. 

We probably needs something like `parseCoverMatchCallAndMatchExpression` as how we are disambiguating `async ()` and `async () => {}`.

Reply:
What is a "primary" expression?

Reply:
Oh I mean https://tc39.es/ecma262/#prod-PrimaryExpression

Reply:
hmm, i'm not super sure about this part. i would expect `match () {}` to be allowed anywhere any other expression is currently allowed (nested inside the pin operator in another match construct, of course, it would only be allowed with parens)

Reply:
Ok, I have open https://github.com/tc39/proposal-pattern-matching/issues/214 and let's continue there.